### PR TITLE
Reject on region

### PR DESCRIPTION
**Pull request by e-koch**
Remove regions in the plane which are smaller than the beam area. This builds off of the more general ```reject_region```, which I also implemented here. I assumed that the roughed-in description meant it was for general handling of 2D rejection criteria.

### CODE REVIEW

File: signal_id/mask.py [hunk 1]
```diff
@@ -320,10 +321,93 @@ def close(self, struct=None, iterations=1):
...
+            Beam object from the cube WCS.
+
+        '''
+
+        self.log_and_backup(self.remove_small_regions)
+
+        # Attempt to get beam area from cube WCS info.
+        if area_threshold is None:
+            if beam is None:
+                beam = Beam.from_fits_header(self.cube.wcs.to_header())
```
Comment (e-koch) [inline signal_id/mask.py]: Ah. The WCS object won't have any beam info... I'll need to rethink this a bit.

Comment (keflavich) [inline signal_id/mask.py]: shouldn't the `cube` have a header?  I don't think you need to recreate one from `wcs`

Comment (e-koch) [inline signal_id/mask.py]: That makes much more sense!

**Comment by e-koch** [comment]
Good to go! Local testing on a ginormous cube was working.

This now includes some minor general fixes for ```RadioMask```. I don't think the existing code has been tested well (has anyone really tried using it yet?).

@keflavich @akleroy @low-sky -- would someone mind having a quick look over the changes before I merge?

File: signal_id/mask.py [hunk 2]
```diff
@@ -97,7 +101,7 @@ def from_file(self, fname, thresh=None, format='fits'):
 
     def from_spec_cube(self, cube, thresh=None):
         self._linked_data = cube
-        self._mask = cube._mask.include
+        self._mask = cube._mask.include()
```
Comment (keflavich) [inline signal_id/mask.py]: can use `cube.mask` now (it's no longer private)

File: signal_id/mask.py [hunk 3]
```diff
@@ -232,10 +236,12 @@ def independent_cubes(self, struct=None):
...
 
     def enable_backup(self):
-        self._implict_backup = True
+        self.is_backup_enabled = True
+        return self
 
     def disable_backup(self):
-        self._implict_backup = False
+        self.is_backup_enabled = False
+        return self
```
Comment (keflavich) [inline signal_id/mask.py]: why is `self` being returned here?  Just curious what the interface is meant to do.  Normally a state change like this would have no return, right?  You might also make `is_backup_enabled` a `property` since it's so simple

File: signal_id/utils.py [hunk 4]
```diff
@@ -138,3 +138,19 @@ def get_pixel_scales(mywcs):
     #scale = np.array([cdelt[0] * (pc[0,0]**2 + pc[1,0]**2)**0.5,
     # cdelt[1] * (pc[0,1]**2 + pc[1,1]**2)**0.5])
     #return abs(scale[0])
+
+
+def get_celestial_axes(wcs):
```
Comment (keflavich) [inline signal_id/utils.py]: It looks like this isn't being used?  Can any of the functions here: http://astropy.readthedocs.org/en/latest/wcs/#subsetting-and-pixel-scales do what you want?  or http://astropy.readthedocs.org/en/latest/api/astropy.wcs.WCS.html#astropy.wcs.WCS.get_axis_types?

**Comment by keflavich** [comment]
Looks good to me. My commentary is nitpicky.  However, it would be good to have an example in the docstrings and/or a documentation entry for this.

File: signal_id/mask.py (hunk 3 above)
Comment (e-koch) [inline signal_id/mask.py]: You're right, this shouldn't be returning anything. I'm going to keep ```is_backup_enabled``` as is for now; there's a bunch of work to be done on the logging and it deserves its own PR.

File: signal_id/utils.py (hunk 4 above)
Comment (e-koch) [inline signal_id/utils.py]: Deleted; it wasn't needed.

**Comment by e-koch** [comment]
Thanks for the comments! I retested the changes and am going to merge.
